### PR TITLE
solveset_complex(0, x) now returns S.Complex

### PR DESCRIPTION
**Pull request by shivamT95**
Fixing issue #9584 .
Also fixed the test for the same.

### CODE REVIEW

File: sympy/solvers/tests/test_solveset.py [hunk 1]
```diff
@@ -588,7 +588,7 @@ def test_solveset_complex_polynomial():
 
 def test_sol_zero_complex():
     # This should return the complex set after it is implemented
-    raises(NotImplementedError, lambda: solveset_complex(0, x))
+    assert solveset_complex(0,x) == S.Complex
```
Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]: put space after comma:
`(0,x)` --> `(0, x)`

Comment (shivamT95) [inline sympy/solvers/tests/test_solveset.py]: done, thanks.

**Comment by leosartaj** [comment]
Ideally, the commit message (first line) should be less or equal to 50 characters.

**Comment by aktech** [comment]
Please keep your commit message to 50 characters or less.  If you have more to say than that, put it in additional paragraphs wrapped to 72 characters. (Learn more)

**Comment by aktech** [comment]
Anyways, +1

**Comment by aktech** [comment]
Same applies for PR titles as well (short & descriptive), I have changed that as of now.

**Comment by shivamT95** [comment]
@leosartaj @aktech, Thanks will keep that in mind! I am still new to these things.

**Comment by aktech** [comment]
> I am still new to these things.

It's absolutely fine, everyone is learning here. :)

File: sympy/solvers/tests/test_solveset.py [hunk 2]
```diff
@@ -588,7 +588,7 @@ def test_solveset_complex_polynomial():
 
 def test_sol_zero_complex():
     # This should return the complex set after it is implemented
-    raises(NotImplementedError, lambda: solveset_complex(0, x))
+    assert solveset_complex(0, x) == S.Complex
```
Comment (leosartaj) [inline sympy/solvers/tests/test_solveset.py]: It would be better to use ``is`` rather than ``==``.

Comment (whitten) [inline sympy/solvers/tests/test_solveset.py]: What does "is" give us that "==" does not ?

Comment (leosartaj) [inline sympy/solvers/tests/test_solveset.py]: ``is`` will return ``True`` if two variables point to the same object, ``==`` if the objects referred to by the variables are equal. As ``S.Complex`` is a singleton, so it's the same object. However, ``==`` will also give the desired result.

Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]: @leosartaj 
Is there a PEP8 standard for this?
As far as I know, we normally use `is` for values like `True` or `False`.

Comment (leosartaj) [inline sympy/solvers/tests/test_solveset.py]: I don't really know. It's just a suggestion. I am fine with ``==`` as well.

**Comment by aktech** [comment]
It's ready to go now, @shivamT95  can you [squash commits](http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git)?

**Comment by shivamT95** [comment]
@aktech Done, Thanks

**Comment by hargup** [comment]
+1

**Comment by aktech** [comment]
Thanks for working on this, It's in.

File: sympy/solvers/tests/test_solveset.py (hunk 2 above)
Comment (gxyd) [inline sympy/solvers/tests/test_solveset.py]: The comment above this could also have been removed.

Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]: Ah! missed that, thanks!

Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]: Here is the Issue: #9620 now.

**Pull request merged**